xbmc/guilib/GUIFontTTF.cpp

```diff
@@ -428,6 +437,28 @@ void CGUIFontTTF::DrawTextInternal(CGraphicContext& context,
       // Offset this line's starting position
       startX -= w;
     }
+    else if (alignment & XBFONT_RIGHT && alignment & XBFONT_TRUNCATED)
+    {
+      // We need to determinate where the text will be truncated on the left
+      // by taking in account that we have also to add ellipses before the text
+      float textWidth = ellipsesWidth;
+      // We need to iterate from the end to the beginning
+      auto itRGliph = glyphs.rbegin();
+      for (; itRGliph != glyphs.rend(); itRGliph++)
```

Comment:
Possible to combine on a single line?
```suggestion
      for (auto itRGliph = glyphs.rbegin(); itRGliph != glyphs.rend(); itRGliph++)
```

Reply:
done

Reply:
itRGlyph please